src/kw_config_loader.sh

```diff
@@ -141,8 +141,22 @@ function parse_configuration()
 # higher level setting definitions to overwrite lower level ones.
 function load_configuration()
 {
+  local -a config_dirs
+  local IFS=:
+  read -ra config_dirs <<< "${XDG_CONFIG_DIRS:-"/etc/xdg"}"
+  unset IFS
+
   parse_configuration "$KW_ETC_DIR/$CONFIG_FILENAME"
-  parse_configuration "$HOME/.kw/$CONFIG_FILENAME"
+
+  # XDG_CONFIG_DIRS is a colon-separated list of directories for config
+  # files to be searched, in order of preference. Since this function
+  # reads config files in a reversed order of preference, we must
+  # traverse it from back to top
+  for ((i = ${#config_dirs[@]} - 1; i >= 0; i--)); do
```

Comment:
I think it will be easier to read this code if you move `${#config_dirs[@]} - 1` outside this `for` loop.
Also, in the comment, could you add one example that justifies this `for`?